src/lib.rs

```diff
@@ -424,6 +424,34 @@ impl<S> StreamUnordered<S> {
         })
     }
 
+    /// Returns an immutable iterator that allows getting a reference to each stream in the set.
+    pub fn iter(&self) -> Iter<'_, S>
```

Comment:
I think it'll probably be pretty confusing for `iter` to include a token, but `iter_mut` not to. Let's instead add this as `iter_with_token` and add a `iter_mut_with_token` as the `&mut` version. Then in the next major version we can get rid of the non-token iterators and just have them all provide tokens instead.